unumpy/_multimethods.py

```diff
@@ -1749,13 +1739,21 @@ def _roll_default(a, shift, axis=None):
 
     axis = _normalize_axis(ndim(a), axis, allow_repeats=True)
 
-    if len(axis) >= len(shift):
-        shift = broadcast_to(shift, (len(axis),))
-    else:
-        axis = broadcast_to(axis, (len(shift),))
+    if not len(axis) == len(shift):
+        if len(shift) == 1:
+            shift = tuple(itertools.repeat(shift[0], len(axis)))
+        elif len(axis) == 1:
+            axis = tuple(itertools.repeat(axis[0], len(shift)))
+        else:
+            raise ValueError("shift or axis must have length of 1.")
```

Comment:
In this context they are both tuples, so I would say must have the same length.

Reply:
Thanks, I wasn't quite sure what the error message should be.